### PR TITLE
doc: improved Saver documentation.

### PR DESCRIPTION
**Pull request by hermet**

### CODE REVIEW

File: inc/thorvg.h [hunk 1]
```diff
@@ -1339,7 +1339,17 @@ class TVG_EXPORT Initializer final
 /**
  * @class Saver
  *
- * @brief A class enabling saving a paint in a binary format.
+ * @brief A class exporting a paint in a specified file data to recover the paint scene later.
```
Comment (mgrudzinska) [inline inc/thorvg.h:1342]: I would write:
A class for exporting a paint object into a specified file, from which to recover the paint data later.

File: inc/thorvg.h [hunk 2]
```diff
@@ -1339,7 +1339,17 @@ class TVG_EXPORT Initializer final
...
+ *
+ * ThorVG provides a feature for exporting & importing paint data, the Saver has a role to export it to a file.
```
Comment (mgrudzinska) [inline inc/thorvg.h:1344]: ThorVG provides a feature for exporting & importing paint data. The purpose of the Saver is to export the data to a file.

File: inc/thorvg.h [hunk 3]
```diff
@@ -1339,7 +1339,17 @@ class TVG_EXPORT Initializer final
...
+ * Basically, this feature is useful when you need to save the composed scene or image from a paint object and replay it later.
+ *
+ * The file format is decided by the extension name(i.e. "*.tvg") while the supported formats depend on the TVG packaging environment.
+ * If it doesn't support the file format, it will return the @c NonSuppport result by the save() method.
```
Comment (mgrudzinska) [inline inc/thorvg.h:1348]: When trying to save a file with an unknown extension or a non-supported format, the @c NonSuppport result is returned by the save() method.

File: inc/thorvg.h [hunk 4]
```diff
@@ -1349,21 +1359,46 @@ class TVG_EXPORT Saver
...
+     * @see Saver::sync()
+     *
+     * @BETA_API
      */
     Result save(std::unique_ptr<Paint> paint, const std::string& path) noexcept;
+
+    /**
+     * @brief Guarantees that the saving task is finished.
+     *
+     * The behavior of the saver will work on sync/async based on the threading setting of the initializer.
```
Comment (mgrudzinska) [inline inc/thorvg.h:1381]: The behavior of the saver will work on a sync/async basis, depending on the threading setting of the Initializer.

File: inc/thorvg.h [hunk 5]
```diff
@@ -1349,21 +1359,46 @@ class TVG_EXPORT Saver
...
+     * Thus if you wish to have a benefit of it, you must call sync() after the save() in the proper delayed time.
```
Comment (mgrudzinska) [inline inc/thorvg.h:1382]: It's not very clear for me from this description when the proper time is, but grammatically it's ok.

File: inc/thorvg.h [hunk 6]
```diff
@@ -1349,21 +1359,46 @@ class TVG_EXPORT Saver
...
+     * Otherwise, you can call sync() immediately.
+     *
+     * @return Result::Success when succeed.
+     * @return Result::InsufficientCondition otherwise.
+     *
+     * @note the asynchronous tasking is depend on the saver module implementation.
```
Comment (mgrudzinska) [inline inc/thorvg.h:1388]: "The" instead of "the"

File: inc/thorvg.h [hunk 7]
```diff
@@ -1339,7 +1339,17 @@ class TVG_EXPORT Initializer final
 /**
  * @class Saver
  *
- * @brief A class enabling saving a paint in a binary format.
+ * @brief A class exporting a paint in a specified file data to recover the paint scene later.
+ *
+ * ThorVG provides a feature for exporting & importing paint data, the Saver has a role to export it to a file.
+ * Basically, this feature is useful when you need to save the composed scene or image from a paint object and replay it later.
```
Comment (mgrudzinska) [inline inc/thorvg.h:1345]: I'd change replay -> recreate

File: inc/thorvg.h [hunk 8]
```diff
@@ -1339,7 +1339,17 @@ class TVG_EXPORT Initializer final
...
+ *
+ * The file format is decided by the extension name(i.e. "*.tvg") while the supported formats depend on the TVG packaging environment.
+ * If it doesn't support the file format, it will return the @c NonSuppport result by the save() method.
+ *
+ * Once you export a paint to the file successfully, you can replay it using the Picture class.
```
Comment (mgrudzinska) [inline inc/thorvg.h:1350]: I'd change replay -> recreate

File: inc/thorvg.h (hunk 1 above)
Comment (hermet) [reply inc/thorvg.h:1342]: updated.

File: inc/thorvg.h (hunk 2 above)
Comment (hermet) [reply inc/thorvg.h:1344]: updated

File: inc/thorvg.h (hunk 3 above)
Comment (hermet) [reply inc/thorvg.h:1348]: updated

File: inc/thorvg.h (hunk 6 above)
Comment (hermet) [reply inc/thorvg.h:1388]: updated

File: inc/thorvg.h (hunk 8 above)
Comment (hermet) [reply inc/thorvg.h:1350]: updated

File: inc/thorvg.h (hunk 7 above)
Comment (hermet) [reply inc/thorvg.h:1345]: updated

**Pull request closed without merging**